### DISCUSSION TITLE
UserWarning: Your `val_dataloader` has `shuffle=True`,it is strongly recommended that you turn this off for val/test/predict dataloaders.

### DISCUSSION THREAD

**Discussion by hiyyg** [Lightning-AI/pytorch-lightning#10771] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
It is strange that my `val_dataloader`'s `shuffle` is set to `False`, but I still get this warning. Any ideas on how to solve this?

**Comment by hiyyg** [comment, marked as answer]
I know what the bug is, the sampler used for val_dataloader is not `SequentialSampler` but it does not do any shuffling.

**Reply by tchaton** [reply]
Yes, this is just a warning as Lightning is trying to help out. We can't really know what custom samplers are doing. Sometimes it is good practice to subclass directly the SequentialSampler to inform of a non random sampling.
